File: basics/anagram.py
first = input("First string: ")
second = input("Second string: ")

# remove punctuation, spaces, and capitalization
no_punc_1 = [i.lower() if i.isalnum() else "" for i in first]
no_punc_2 = [i.lower() if i.isalnum() else "" for i in second]
#            ^ removing capitalization
#                         ^ only letters and numbers; alternatively, i.lower() in "abcdefghijklmnopqrstuvwxyz1234567890"
#                                             ^ applies this selection to each character

# no need for [] square brackets because Python knows what you mean
# square brackets are fine though

# check if you can arrange one into the other (is it an anagram?)

if sorted(no_punc_1)==sorted(no_punc_2):
    print("they are anagrams")
else:
    print("they are not anagrams")

File: basics/anagram.cpp
#include <iostream>
#include <algorithm>
using namespace std;
int main() {
    string first,second,
    no_punc_1   ,no_punc_2;
    cout << "First string: ";
    getline(cin,first);
    cout << "Second string: ";
    getline(cin,second);
    // remove punctuation, spaces, and capitalization
    for (char c: first) {
        // if it's uppercase, make it lowercase and add it
        if('A' <= c and c <= 'Z') {
            c += 'a' - 'A'; // 'A' < 'a'
            no_punc_1 += c;
        }
        if( ('a' <= c and c <= 'z') or
            ('0'<= c and c <= '9') )
            no_punc_1 += c;
    }
    for (char c: second) {
        // if it's uppercase, make it lowercase and add it
        if ('A' <= c and c <= 'Z') {
            c += 'a' - 'A'; // 'A' < 'a'
            no_punc_2 += c;
        }
        if( ('a' <= c and c <= 'z') or
            ('0' <= c and c <= '9') ) 
            no_punc_2 += c;
    }
    sort(no_punc_1.begin(),no_punc_1.end());
    sort(no_punc_2.begin(),no_punc_2.end());
    
    cout << "they are ";
    if (no_punc_1 != no_punc_2)
        cout << "not ";
    cout << "anagrams" << endl;
}